src/gui/sdlmain.cpp

```diff
@@ -3205,13 +3210,17 @@ static void set_output(Section* sec, const bool wants_aspect_ratio_correction)
 		if (!set_default_window_mode()) {
 			LOG_WARNING("OPENGL: Could not create OpenGL window, "
 			            "using 'texture' output mode");
+
 			sdl.want_rendering_backend = RenderingBackend::Texture;
+			set_texture_output_setting();
```

Comment:
Good pickup, thanks. I will take `sdl.interpolation_mode` into consideration too here, and set `texture` or `texturenb` accordingly. The log message itself is misleading;, we should log `texture` or `texturenb` when falling back to the texture backend from `opengl` or `openglnb`, respectively.

Only the log message was guiding my change, so thanks for looking into it more deeply 🎉 

> Again, I'm not sure what your intent was here. In this case, the config was already set to a valid value but you're updating it based on a runtime error.

Yes, correct that is what's happening.

One can set a config setting to a valid value, e.g. you can set `cms = off`. That's valid. But if you enable the Game Blaster with `sbtype = gb`, the CMS _must_ be enabled as there is no Game Blaster hardware without CMS. So this case, we log an error and notify the user about the forced setting change.

<img width="929" alt="image" src="https://github.com/dosbox-staging/dosbox-staging/assets/698770/a36efb74-53f9-43e7-9a69-e9225171f3e1">

<img width="840" alt="image" src="https://github.com/dosbox-staging/dosbox-staging/assets/698770/f85ffe27-37f4-494f-aa97-26ceac1fbab7">

Naturally, it would be rather confusing to leave the `cms` setting on `off`, don't you agree?

To use a GUI analogy, in more complex scenarios other GUI elements will be auto-reset for you. E.g., in this case selecting the "Game Blaster" from a dropdown would auto-tick the CMS option. This is pretty standard.

> Is the config value meant to be "this is what the user wants" or "this is what we're actually running"?

Yes, the latter—the currently active setting.

Reply:
Also, conflating the interpolation method into the rendering backend's name is terrible; we intended to decouple these years ago, but somehow, it never happened. I thought we had a ticket for it, but nope.

We should have something like this:

```ini
output = texture | opengl
texture_interpolation = bilinear | nearest
```

Forcing nearest for OpenGL is just silly, the shaders should have full control over that, plus I can't see a legit use-case for having "nearest" for OpenGL. I'd much rather introduce a `nearest` shader in addition to `none` (which does bilinear, so we could alias `none` to `bilinear` as well).